feat: Add support for filter/group by country

rafaeelaudibert opened the pull request:
Similar to cohorts, but much simpler, requested by a customer :)

Closes #33500

rafaeelaudibert committed:
feat: Add support for filter/group by country

Similar to cohorts, but much simpler, requested by a customer :)

rafaeelaudibert committed:
docs: Add comprehensive guide for implementing new Revenue Analytics filter/group by option

This commit introduces a detailed guide on adding a new filter and group by option for Revenue Analytics, specifically the "country" option. The guide covers necessary changes across frontend schema, taxonomy definitions, database views, HogQL property mapping, API endpoints, and frontend component integration. It also includes build commands and a summary of files to modify for successful implementation.

AI-generated of course, and so was this commit

rafaeelaudibert commented on products/revenue_analytics/backend/hogql_queries/revenue_analytics_insights_query_runner.py:
```diff
@@ -178,6 +180,92 @@ def _get_subquery_by_product(self) -> ast.SelectQuery | None:
 
         return query
 
+    def _get_subquery_by_country(self) -> ast.SelectQuery | None:
```
Yes, this is a bunch of copy-paste, https://github.com/PostHog/posthog/issues/33502 is supposed to fix this with a much more generic interface

github-actions[bot] committed:
Update UI snapshots for `chromium` (7)

lricoy commented on products/revenue_analytics/guides/new_option_for_filters_and_group_by.md:
I love the guides, keep them coming!

lricoy reviewed (approved):
SHIP IT

jabahamondes reviewed (approved):
LGTM 🚀 , there are some good bot comments to apply btw

rafaeelaudibert commented on products/revenue_analytics/backend/api.py:
```diff
@@ -57,6 +57,21 @@ def values(self, request: Request, **kwargs):
                 select_from=ast.JoinExpr(table=customer_selects_union),
                 order_by=[ast.OrderExpr(expr=ast.Field(chain=["cohort"]), order="ASC")],
             )
+        elif key == "country":  # All countries available from revenue analytics
+            revenue_selects = revenue_selects_from_database(database)
+            customer_selects: list[ast.SelectQuery] = [
+                cast(ast.SelectQuery, select[REVENUE_SELECT_OUTPUT_CUSTOMER_KEY])
+                for select in revenue_selects.values()
+                if select[REVENUE_SELECT_OUTPUT_CUSTOMER_KEY] is not None
+            ]
+            customer_selects_union = ast.SelectSetQuery.create_from_queries(customer_selects, set_operator="UNION ALL")
+
+            query = ast.SelectQuery(
+                select=[ast.Alias(alias="country", expr=ast.Field(chain=["address", "country"]))],
+                distinct=True,
+                select_from=ast.JoinExpr(table=customer_selects_union),
+                order_by=[ast.OrderExpr(expr=ast.Field(chain=["country"]), order="ASC")],
+            )
```
Yeah, I know, #33502 will solve that - by EOD hopefully

rafaeelaudibert commented on products/revenue_analytics/guides/new_option_for_filters_and_group_by.md:
```diff
@@ -0,0 +1,324 @@
+
+Adding a new filter/breakdown option in Revenue Analytics requires changes across multiple layers:
+
+1. **Frontend schema definitions** - Define the new property type
+1. **Taxonomy system** - Define filter metadata for the UI
+1. **Database views** - Ensure required fields are available
+1. **HogQL property handling** - Map the property to database fields
+1. **API endpoints** - Add value fetching for filter dropdowns
+1. **Query runners** - Implement groupBy logic for breakdowns
+1. **Frontend components** - Add UI controls and feature flags
```
1. is better for maintanability, markdown renderers know how to render this properly

rafaeelaudibert committed:
fix: Fix robot typos caught by another robot

rafaeelaudibert committed:
refactor: Simplify revenue analytics values API

rafaeelaudibert committed:
refactor: Small simplification for new `country` field

Accessing it as a map doesnt work because JSON isnt as well supported as I thought, so let's do a JSONExtractString call on the view instead

rafaeelaudibert committed:
fix: Fix mypy types

rafaeelaudibert committed:
fix: Allow events to be selected again

This broke on the recent reorganization of base currency, fixing it in this PR to expedite releasing it

rafaeelaudibert committed:
refactor: Remove useless stories

This isnt needed, we have a more global config one
